enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,398 @@
+---
+title: host-network-configuration
+authors:
+  - "@russellb"
+reviewers:
+  - “@derekwaynecarr”
+  - “@crawford”
+  - “@cgwalters”
+approvers:
+  - “@celebdor”
+  - “@bcrochet”
+  - “@squeed”
+  - “@danwinship”
+  - “@dcbw”
+  - “@knobunc”
+  - “@abhinavdahiya”
+  - “@sdodson”
+  - “@yboaron”
+  - “@qinqon”
+  - “@phoracek”
+  - “@EdDev”
+  - “@ashcrow”
+  - “@dhellmann”
+  - “@hardys”
+  - “@stbenjam”
+  - “@dustymabe”
+  - “@miabbott”
+  - “@dougbtv”
+  - “@eparis”
+  - “@markmc”
+  - “@jwforres”
+creation-date: 2020-06-30
+last-updated: 2020-06-30
+status: implemented
+see-also:
+  - "/enhancements/rhcos/static-networking-enhancements.md"
+  - "/enhancements/network/20190903-SRIOV-GA.md"
+  - "/enhancements/network/host-level-openvswitch.md"
+  - "/enhancements/baremetal/baremetal-provisioning-config.md"
+  - "/enhancements/baremetal/baremetal-provisioning-optional.md"
+---
+
+# Host Network Configuration
+
+## Summary
+
+This document serves as a high level overview of host network configuration for
+OpenShift 4.x clusters.  Host network configuration spans multiple features in
+multiple components.  This enhancement aims to describe use cases and point to
+various references for current or future solutions that help satisfy those use
+cases.
+
+## Motivation
+
+Configuring host network devices on an OpenShift cluster looks a bit different
+than a standalone Linux host.  This document aims to be a starting reference
+for learning more about how this works right now and where to look to learn
+more.
+
+### Goals
+
+* Document the current configuration interfaces available for host networking
+* Provide references to other proposals
+
+### Non-Goals
+
+* Proposing new enhancements.  Instead, those should be covered in other
+  enhancements that this one may reference.
+
+
+## Use Cases
```

Comment:
So this talks about some of the features that are needed, but it doesn't explain _why_ they're needed, which would help in understanding the problem space. Why would the customer's environment be set up so that we need to use a VLAN? Why would someone want to manually configure networking on each host rather than setting up a DHCP server to do it?

Reply:
Yeah, this section is pretty weak.  I tried to add a bit more text to each use case listed, but more could certainly be added here over time.

Reply:
I think it would be helpful to outline common configurations and how they would be achieved today - it might also help highlight the issues around interfaces, e.g there are multiple partially overlapping interfaces the user gets exposed to (dracut, coreos-install, ignition, nmstate)

Reply:
Agree, more realistic examples would be great.  I view this as a living document where we could submit enhancements like this over time as we come across good examples.